.github/workflows/R-CMD-check.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: R-CMD-check
+
+on:
+  workflow_call:
+
+jobs:
+  R-CMD-check:
+    runs-on: ${{ matrix.config.os }}
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: macOS-latest,   r: 'release'}
+          - {os: windows-latest, r: 'release'}
+          - {os: ubuntu-latest,  r: 'release'}
+          - {os: ubuntu-latest,  r: 'devel', http-user-agent: 'release'}
+          - {os: ubuntu-latest,  r: 'oldrel-1'}
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: r-lib/actions/setup-pandoc@v2
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ matrix.config.r }}
+          http-user-agent: ${{ matrix.config.http-user-agent }}
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rcmdcheck
+          needs: check
+
+      - uses: r-lib/actions/check-r-package@v2
+        with:
+          upload-snapshots: true
+          error-on: '"note"'
```

Comment:
I like this, but not all our repos use this FYI

Reply:
I like the spirit of this, but not sure if I want it to be the default setting...

Reply:
This is why there's the `do-r-cmd-check` input.
https://github.com/RMI-PACTA/actions/blob/b15b6dd39f85327fbba8ed15a27bd1c2641d100a/.github/workflows/R.yml#L5-L10

Reply:
If we prefer the default to be false on this, I'm fine with that.

Reply:
`default: false` allows the R CMD check to run but without the "error on note"?

Reply:
`default: false` would disable R CMD CHECK entirely

Reply:
We can add a pass-through input for the error level, but I'm generally inclined to leave it at `NOTE` since those are usually signs that something is an issue with our code, and are almost always addressable.

If we make a passthrough, I'd probably make the default `note`, and let individual repos dial it back to `warning` if they have notes that are irresolvable.

Reply:
Passthrough input added in cb79309